gluten-data/src/main/java/org/apache/gluten/memory/arrow/memory/ArrowNativeMemoryManager.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gluten.memory.arrow.memory;
+
+import org.apache.arrow.dataset.jni.NativeMemoryPool;
+import org.apache.spark.util.TaskResource;
+import org.apache.spark.util.TaskResources;
+import org.apache.spark.util.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArrowNativeMemoryManager implements TaskResource {
```

Comment:
NIT: I think it's fine to use class name `ArrowNativeMemoryPool` and package name `org.apache.gluten.memory.arrow.pool`.

Since `org.apache.gluten.memory.arrow.memory` looks a little bit confusing as package name.